crypto/objects/obj_dat.c

```diff
@@ -128,7 +128,7 @@ static unsigned long added_obj_hash(const ADDED_OBJ *ca)
     a = ca->obj;
     switch (ca->type) {
     case ADDED_DATA:
-        ret = a->length << 20L;
+        ret = (unsigned long)a->length << 20UL;
```

Comment:
It's only the left-hand side of `<<` that needs to be promoted

Reply:
Yeah, but changing the RHS as well is OK.